docs: add new Layout Functions: Center, Column and Row.. (#516)
ZupIT/beagle-docs: content/en/api/components/layout/column.md

---
title: Column
weight: 338
description: You will find here the description of the Column function and its attributes details
---

---

## What is it?

> `Column` is a function that displays your children in a vertical matrix.

Reviewer: ```suggestion
`Column` is a function to help you organize the layout of your view. It  creates a `Container` that displays its content in a vertical line. This is equivalent to create a `Container` with `flexDirection` equal to `COLUMN`.
```

See how the structure is represented:

| **Attribute** | **Type**  | Required | **Definition** |
| :----------- | :------------------------------------------------------------- | :---------: | :---------------------------------------------------------------------------------------------------------------- |

> | children     | List &lt;[ServerDriven Component]({{< ref path="/api/components" lang="pt" >}})&gt; |            | The components below this component in the tree.

Reviewer: ```suggestion
| children     | List &lt;[ServerDriven Component]({{< ref path="/api/components" lang="pt" >}})&gt; |            | The content of this Container, i.e. its children.
```

"below" is not the right word here, since it would be equivalent to the mathematical term "depth" of a node in a tree. A node in a tree can have a higher value for depth than another and not be its child.

> | context      | [ContextData]({{< ref path="/api/context/" lang="pt" >}})                           |             | It is the [**context**]({{< ref path="/api/context" lang="en" >}}) in this widget.                                                    |

Reviewer: ```suggestion
| context      | [ContextData]({{< ref path="/api/context/" lang="pt" >}})                           |             | The [**context**]({{< ref path="/api/context" lang="en" >}}) linked to this widget.                                                    |
```

| onInit       | List&lt;[**Action**]({{< ref path="/api/actions" lang="en" >}})&gt; |  | It is a parameter that allows you to define a list of actions to be performed when the Widget is displayed. |
| styleId   | String                                                |             | Refers to a locally configured style that it will be applied on the container. |

> | reverse   | Boolean                                                |             | Changes the order of items. Default is `false`. |

Reviewer: ```suggestion
| reverse   | Boolean                                                |             | Displays its elements from bottom to top. Default is `false`. |
```
